xbmc/platform/posix/PosixTimezone.cpp

```diff
@@ -195,44 +197,65 @@ void CPosixTimezone::SetTimezone(const std::string& timezoneName)
 
 std::string CPosixTimezone::GetOSConfiguredTimezone()
 {
-   char timezoneName[255];
+  std::string timezoneName;
 
-   // try Slackware approach first
-   ssize_t rlrc = readlink("/etc/localtime-copied-from"
-                           , timezoneName, sizeof(timezoneName)-1);
+  // try Slackware approach first
+  timezoneName = ReadFromLocaltime("/etc/localtime-copied-from");
 
-   // RHEL and maybe other distros make /etc/localtime a symlink
-   if (rlrc == -1)
-     rlrc = readlink("/etc/localtime", timezoneName, sizeof(timezoneName)-1);
+  // then check for webOS
+  if (timezoneName.empty())
+    timezoneName = ReadFromLocaltime("/var/luna/preferences/localtime");
 
-   if (rlrc != -1)
-   {
-     timezoneName[rlrc] = '\0';
-
-     char* p = strrchr(timezoneName,'/');
-     if (p)
-     { // we want the previous '/'
-       char* q = p;
-       *q = 0;
-       p = strrchr(timezoneName,'/');
-       *q = '/';
-       if (p)
-         p++;
-     }
-     return p;
-   }
+  // RHEL and maybe other distros make /etc/localtime a symlink
+  if (timezoneName.empty())
+    timezoneName = ReadFromLocaltime("/etc/localtime");
```

Comment:
this is how systemd configures the timezone, so it should work on any distro with systemd.
I'd make this the first option we try

Reply:
`/etc/localtime` points to `/var/luna/preferences/localtime` on webOS.  So we'd retrieve `preferences/localtime` if it was tried first

Reply:
don't you already use readlink? And that's what was used before as well

Reply:
Oh sorry, I meant `realpath`

Reply:
that should work as well. Quickly tested on ubuntu

Reply:
Do you think it's better to change only to `realpath` and keep the rest of the code or also apply the refactoring as well and adapting that to use `realpath` too?